### PR TITLE
Bump required pytest version to 4.6

### PR DESCRIPTION
**Pull request by Zalathar**
Motivated by #2863, but also a fine idea in general since 4.6 is the earliest version semi-supported by the pytest maintainers.

### CODE REVIEW

**Comment by Zalathar** [comment]
I'm having some trouble with `test_prints_statistics_given_option_under_xdist_with_junitxml` (`test_statistics.py`) under 4.6, so I figured I'd post this as-is to get a second opinion.

File: hypothesis-python/tox.ini
Comment (Zalathar) [inline hypothesis-python/tox.ini:78]: I bumped pytest-xdist to the latest version that supports pytest 4.x.

Not sure if pytest-forked still needs to be pinned; pytest-xdist 1.34 seems to work fine with the latest release.

File: hypothesis-python/src/hypothesis/extra/pytestplugin.py
Comment (Zalathar) [inline hypothesis-python/src/hypothesis/extra/pytestplugin.py:57]: Since I was already updating this message, I took the liberty of making some small editorial tweaks.

**Comment by Zalathar** [comment]
I assume the test failure has some relation to #2743, since that's when the test was added.

**Comment by Zalathar** [comment]
OK, after some investigation it looks like #2743 never actually worked on non-recent versions of pytest.

I'm going to bisect to find a reasonable minimum version for the offending test, and not worry too much about it being broken on earlier versions than that, since we already accepted it being broken for <4.6.

**Comment by Zalathar** [comment]
The new cutoff for that test is pytest <5.4.0, since that's when it seems to magically start working.

**Review by Zac-HD — APPROVED** [review]
Looks good to me, with a few tweaks to the release notes:

- behaviour change = minor release, not patch
- mention that 4.6 is the oldest version of pytest still accepting bugfixes
- I'd use command-line arguments such as --hypothesis-show-statistics rather than the database thing, as a user-visible feature. Current text is fine too.

Happy for you to merge this when ready 😁

**Comment by Zalathar** [comment]
I don't think I can merge this myself, since it's waiting for “check-pytest43” which will never report completion (as it's been renamed by this change).

**Comment by Zac-HD** [comment]
Ah, of course - I've updated the required statuses now 😁

**Pull request merged**
